redecs_agent/redecs_agent.go

```diff
@@ -60,58 +55,9 @@ func reportServiceActive(serviceName string) error {
 	return err
 }
 
-// This reports an instance as inactive. It removes it from the redecs:service_pings set
-// and publishes a redecs:service_channel event.
-func reportServiceInactive(serviceName string) error {
-	log.Debugf("Reporting %s as inactive with IP %s", serviceName, configuration.LocalIp)
-	value := serviceName + "_" + configuration.LocalIp
-	// remove SERVICENAME_IP from redecs:service_pings
-	err := redisClient.ZRem("redecs:service_pings", value).Err()
-	if err != nil {
-		return err
-	}
-	// notify the channel that this service has been reported active
-	err = redisClient.Publish("redecs:service_channel", "-"+value).Err()
-	return err
-}
-
-// Gets the service name from a Docker container environment variable, if present.
-func getServiceName(container *docker.Container) string {
-	// One of the environment variables should be SERVICE_<port>_NAME = <name of the service>
-	// We look for this environment variable doing a split in the "=" and another one in the "_"
-	// So envEval = [SERVICE_<port>_NAME, <name>]
-	// nameEval = [SERVICE, <port>, NAME]
-	for _, env := range container.Config.Env {
-		envEval := strings.Split(env, "=")
-		nameEval := strings.Split(envEval[0], "_")
-		if len(envEval) == 2 && len(nameEval) == 3 && nameEval[0] == "SERVICE" && nameEval[2] == "NAME" {
-			if _, err := strconv.Atoi(nameEval[1]); err == nil {
-				return envEval[1]
-			}
-		}
-	}
-	return ""
-}
-
-func processContainers() {
-	log.Debug("Listing active containers...")
-
-	// only get running ones
-	containers, err := dockerClient.ListContainers(docker.ListContainersOptions{All: false})
-	// bail out completely if we can't list containers.
-	logErrorAndFail(err)
-	for _, containerEntry := range containers {
-		// have to inspect to get the environment variable
-		container, err := dockerClient.InspectContainer(containerEntry.ID)
-		// retry logic?
-		logErrorAndFail(err)
-		serviceName := getServiceName(container)
-		if serviceName != "" {
-			reportServiceActive(serviceName)
-		}
-	}
-
-	log.Debug("Done checking active containers")
+// Gets the service name from the SERVICE_NAME environment variable.
+func getServiceName() string {
+	return os.Getenv("SERVICE_NAME")
 }
 
 func main() {
```

Comment:
`err = errors.New(fmt.Sprintf("Usage: %s [Redis host] {host IP (for testing)}\n", os.Args[0]))`

should say current IP instead?